File: Windows/GameControllerSimulator/GameControllerSimulator/Bean/DeviceInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace GameControllerSimulator.Bean
{

    public class DeviceInfo
    {
        [JsonPropertyName("os_version")]
        public string OsVersion { get; set; } = string.Empty;

        [JsonPropertyName("sdk")]
        public string Sdk { get; set; } = string.Empty;

        [JsonPropertyName("brand")]
        public string Brand { get; set; } = string.Empty;

        [JsonPropertyName("manufacturer")]
        public string Manufacturer { get; set; } = string.Empty;

        [JsonPropertyName("model")]
        public string Model { get; set; } = string.Empty;

        [JsonPropertyName("device")]
        public string Device { get; set; } = string.Empty;

        [JsonPropertyName("product")]
        public string Product { get; set; } = string.Empty;

        [JsonPropertyName("board")]
        public string Board { get; set; } = string.Empty;

        [JsonPropertyName("hardware")]
        public string Hardware { get; set; } = string.Empty;

        [JsonPropertyName("codename")]
        public string Codename { get; set; } = string.Empty;

        [JsonPropertyName("build_id")]
        public string BuildId { get; set; } = string.Empty;

        [JsonPropertyName("fingerprint")]
        public string Fingerprint { get; set; } = string.Empty;

        [JsonPropertyName("supported_abis")]
        public string SupportedAbis { get; set; } = string.Empty;
    }
}

File: Android/GameControllerSimulator/app2/src/main/java/cn/maoyanluo/gamecontrollersimulator2/bean/DeviceInfo.kt
package cn.maoyanluo.gamecontrollersimulator2.bean

import android.os.Build
import com.google.gson.annotations.SerializedName

class DeviceInfo {
    @SerializedName("os_version")
    val osVersion: String = "Android ${Build.VERSION.RELEASE}"

    @SerializedName("sdk")
    val sdk: String = "${Build.VERSION.SDK_INT}"

    @SerializedName("brand")
    val brand: String = "${Build.BRAND}"

    @SerializedName("manufacturer")
    val manufacturer: String = "${Build.MANUFACTURER}"

    @SerializedName("model")
    val model: String = "${Build.MODEL}"

    @SerializedName("device")
    val device: String = "${Build.DEVICE}"

    @SerializedName("product")
    val product: String = "${Build.PRODUCT}"

    @SerializedName("board")
    val board: String = "${Build.BOARD}"

    @SerializedName("hardware")
    val hardware: String = "${Build.HARDWARE}"

    @SerializedName("codename")
    val codename: String = "${Build.VERSION.CODENAME}"

    @SerializedName("build_id")
    val buildId: String = "${Build.ID}"

    @SerializedName("fingerprint")
    val fingerprint: String = "${Build.FINGERPRINT}"

    @SerializedName("supported_abis")
    val supportedAbis: String = Build.SUPPORTED_ABIS.joinToString()

}
